src/Evolution/Systems/GeneralizedHarmonic/Initialize.hpp

```diff
@@ -0,0 +1,217 @@
+        gr::Tags::TraceSpatialChristoffelFirstKindCompute<Dim, frame,
+                                                          DataVector>,
+        GeneralizedHarmonic::Tags::ExtrinsicCurvatureCompute<Dim, frame>,
+        GeneralizedHarmonic::Tags::TraceExtrinsicCurvatureCompute<Dim, frame>,
+        GeneralizedHarmonic::Tags::ConstraintGamma0Compute<Dim, frame>,
+        GeneralizedHarmonic::Tags::ConstraintGamma1Compute<Dim, frame>,
+        GeneralizedHarmonic::Tags::ConstraintGamma2Compute<Dim, frame>>;
+
+    return std::make_tuple(
+        Initialization::merge_into_databox<InitializeGhAnd3Plus1Variables,
+                                           db::AddSimpleTags<>, compute_tags>(
+            std::move(box)));
+  }
+};
+
+template <size_t Dim>
+struct InitializeGauge {
+  using frame = Frame::Inertial;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    // compute initial-gauge related quantities
+    const auto mesh = db::get<::Tags::Mesh<Dim>>(box);
+    const size_t num_grid_points = mesh.number_of_grid_points();
+    const auto& lapse = get<gr::Tags::Lapse<DataVector>>(box);
+    const auto& dt_lapse = get<::Tags::dt<gr::Tags::Lapse<DataVector>>>(box);
+    const auto& deriv_lapse = get<
+        ::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<Dim>, frame>>(
+        box);
+    const auto& shift = get<gr::Tags::Shift<Dim, frame, DataVector>>(box);
+    const auto& dt_shift =
+        get<::Tags::dt<gr::Tags::Shift<Dim, frame, DataVector>>>(box);
+    const auto& deriv_shift =
+        get<::Tags::deriv<gr::Tags::Shift<Dim, frame, DataVector>,
+                          tmpl::size_t<Dim>, frame>>(box);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<Dim, frame, DataVector>>(box);
+    const auto& trace_extrinsic_curvature =
+        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(box);
+    const auto& trace_christoffel_last_indices =
+        get<gr::Tags::TraceSpatialChristoffelFirstKind<Dim, frame, DataVector>>(
+            box);
+
+    // compute the initial gauge source function
+    const auto initial_gauge_h = GeneralizedHarmonic::gauge_source<Dim, frame>(
+        lapse, dt_lapse, deriv_lapse, shift, dt_shift, deriv_shift,
+        spatial_metric, trace_extrinsic_curvature,
+        trace_christoffel_last_indices);
+    // set time derivatives of InitialGaugeH = 0
+    // NOTE: this will need to be generalized to handle numerical initial data
```

Comment:
Could you file an issue about this so we have a record outside of the file?

Reply:
#1708